lib/mixins/template-stamp.html

```diff
@@ -154,8 +150,7 @@
        *         templateInfo: {object}, // nested template metadata
        *         // Metadata to allow efficient retrieval of instanced node
        *         // corresponding to this metadata
-       *         parentInfo: {number},   // reference to parent nodeInfo>
-       *         parentIndex: {number},  // index in parent's `childNodes` collection
+       *         parentInfo: {Array},   // indices to traverse to this node from the template root
```

Comment:
Unfortunately, this is a breaking change. I don't think we can accept this until Polymer 3.0.

Reply:
It may be possible to factor `findTemplateNode` such that it is still recursive, but able to operate either on a `parentInfo` tree (for backward compatibility) or a `parentIndex` array (for efficient serialization).  We can consider this if we move forward with https://github.com/Polymer/polymer/pull/4782 (or a non-breaking refactoring of it).  In that way, the runtime code would still generate backward-compatible `parentInfo` trees (i.e. https://github.com/Polymer/polymer/pull/4853/files#diff-84d426316fb9e8502af8f5eed8675b40L286 would be reverted), and the build-time tool would transform that into an array of indices.

Reply:
It could certainly work by being able to parse both recursive notation as well as the array notation, but I don't think it would be good to land that before Polymer 3. Since the build-time tool is likely not ready before 3 is ready, landing it right now increases the Polymer code size and adds more conditions.